sdk/eventhub/Azure.Messaging.EventHubs.Experimental/src/Producer/IdempotentProducer.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Producer;
+
+namespace Azure.Messaging.EventHubs.Experimental
```

Comment:
I don't think we need "Experimental" in the namespace. I would just put this type in the same namespace as the producer.

Reply:
My reason for splitting this type out was so that all of the experimental types were in the same namespace.

The challenge with that is going to be the `internal` class shadows.   Things like `PartitionPublishingOptions` which exist in the producer namespace as `internal` but have to be redefined here to be made `public`.    If I try to put them all in the same namespace, I'm pretty sure that they'll conflict and won't compile.

Thoughts?